From f3ff50f68eb74bdf8d42e73de1fe478309d5f0dd Mon Sep 17 00:00:00 2001
From: Tejas Patel <tejas.patel1@einfochips.com>
Date: Wed, 19 Jan 2022 13:23:52 +0530
Subject: [PATCH 08/13] arch: arm64: boot: dts: Add DT binding node for AP1302

This patch adds DT binding node for AP1302 in Thor96 device tree file.

Change-Id: I6f2008fdbaf70850d1afd76938623b46fd34eace
Signed-off-by: Tejas Patel <tejas.patel1@einfochips.com>
Signed-off-by: Tanvi Chauhan <tanvi.chauhan@einfochips.com>
---
 .../dts/freescale/imx8mq-thor96-ap1302.dts    | 67 +++++++++++++++++++
 1 file changed, 67 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mq-thor96-ap1302.dts

diff --git a/arch/arm64/boot/dts/freescale/imx8mq-thor96-ap1302.dts b/arch/arm64/boot/dts/freescale/imx8mq-thor96-ap1302.dts
new file mode 100644
index 000000000..b96fcf0e3
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mq-thor96-ap1302.dts
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright 2019 NXP.
+ */
+
+#include "imx8mq-thor96.dts"
+
+&ov5640_mipi2 {
+	status = "disabled";
+};
+
+&ov5640_mipi {
+	status = "disabled";
+};
+
+&i2c3 {
+	ap1302_mipi: ap1302@3d {
+		compatible = "onnn,ap1302";
+		reg = <0x3d>;
+
+		pinctrl-0 = <&pinctrl_csi>;
+		clocks = <&clk IMX8MQ_CLK_CLKO2>;
+		clock-names = "xclk";
+		assigned-clocks = <&clk IMX8MQ_CLK_CLKO2>,
+				  <&clk IMX8MQ_CLK_CLKO2>;
+		assigned-clock-parents = <&clk IMX8MQ_SYS2_PLL_200M>;
+		assigned-clock-rates = <0>, <20000000>;
+
+		reset-gpios = <&gpio3 24 GPIO_ACTIVE_LOW>;
+		standby-gpios = <&gpio2 7 GPIO_ACTIVE_HIGH>;
+		i2csel-gpios = <&gpio2 8 GPIO_ACTIVE_HIGH>;
+		isptrig-gpios = <&gpio3 12 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+
+		port {
+			ap1302_ep: endpoint {
+				remote-endpoint = <&mipi1_sensor_ep>;
+				data-lanes = <1 2 3 4>;
+			};
+		};
+
+		sensors {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			onnn,model = "onnn,ar0430";
+			sensor@0 {
+				reg = <0>;
+			};
+		};
+	};
+};
+
+&mipi1_sensor_ep {
+	remote-endpoint = <&ap1302_ep>;
+	data-lanes = <1 2 3 4>;
+	csis-wclk;
+};
+
+&iomuxc {
+	pinctrl_csi: csi_grp {
+		fsl,pins = <
+			MX8MQ_IOMUXC_SD1_DATA5_GPIO2_IO7	0x19
+			MX8MQ_IOMUXC_SAI5_RXD3_GPIO3_IO24	0x19
+			MX8MQ_IOMUXC_GPIO1_IO15_CCMSRCGPCMIX_CLKO2      0x59
+		>;
+	};
+};
-- 
2.17.1

